src/notifiers/slack_notifier.py

```diff
@@ -65,33 +46,27 @@ def _handle_response(response: Dict[str, Any], channels: List[str]) -> None:
 
     def _send(self, message: SlackMessage) -> Dict[str, Any]:
         response = requests.post(
-            url=self._api_url,
+            url=self._notifier_config.api_url,
             headers=self._build_headers(),
-            json=self._build_payload(message),
+            json=message.to_dict(),
             timeout=10,
         )
         response.raise_for_status()
         return dict(response.json())
 
     def _build_headers(self) -> Dict[str, str]:
-        credentials = b64encode(f"{self._username}:{self._token}".encode("utf-8")).decode("utf-8")
+        credentials = b64encode(
+            f"{self._notifier_config.username}:{self._notifier_config.token}".encode("utf-8")
+        ).decode("utf-8")
         return {"Content-Type": "application/json", "Authorization": f"Basic {credentials}"}
 
-    @staticmethod
-    def _build_payload(message: SlackMessage) -> Dict[str, Any]:
-        return {
-            "channelLookup": {
-                "by": "slack-channel",
-                "slackChannels": message.channels,
-            },
-            "messageDetails": {
-                "text": message.header,
-                "attachments": [
-                    {
-                        "color": message.color,
-                        "title": message.title,
-                        "text": message.text,
-                    }
-                ],
-            },
-        }
+    def apply_filters(self, findings: Set[Finding]) -> Set[Finding]:
+        return FindingsFilter().do_filter(findings, self._filters_config)
+
+    def apply_mappings(self, findings: Set[Finding]) -> List[SlackMessage]:
+        return NotificationMapper().do_map(findings, self._mappings_config, self._org_client)
+
+    # this method should replace send_messages()
```

Comment:
Not really sure what's meant by this comment. In what context?

Reply:
Ignore... just me ~thinking~ writing out loud